### PR TITLE
ast, parser: support syntax for "EXPLAIN FORMAT = stringlit"

### PR DESCRIPTION
**Pull request by zz-jason**
to #4279 

The default format is `ROW`, and we can add another format like `DOT`(#3933 ), and print a dot graph

### CODE REVIEW

**Comment by zz-jason** [comment]
@winkyao PTAL

File: parser/parser.y [hunk 1]
```diff
@@ -1964,7 +1964,17 @@ ExplainStmt:
 	}
 |	ExplainSym ExplainableStmt
 	{
-		$$ = &ast.ExplainStmt{Stmt: $2.(ast.StmtNode)}
+		$$ = &ast.ExplainStmt{
+			Stmt:	$2.(ast.StmtNode),
+			Format: "row",
+		}
+	}
+|	ExplainSym "FORMAT" "=" stringLit ExplainableStmt
```
Comment (XuHuaiyu) [inline parser/parser.y]: please add test for this

File: ast/misc.go [hunk 2]
```diff
@@ -84,7 +84,8 @@ type AuthOption struct {
 type ExplainStmt struct {
 	stmtNode
 
-	Stmt StmtNode
+	Stmt   StmtNode
+	Format string
```
Comment (winkyao) [inline ast/misc.go]: Could we use "const" instead of a string?

Comment (zz-jason) [reply ast/misc.go]: no we can't, when a input explain format is not supported, we should return an error like
`ERROR 1791 (HY000): Unknown EXPLAIN format name: '%s'`

File: parser/parser.y (hunk 1 above)
Comment (zz-jason) [reply parser/parser.y]: added

**Pull request merged**